core/src/main/java/feast/core/model/FeatureSet.java

```diff
@@ -121,6 +121,53 @@ private String getProjectName() {
     }
   }
 
+  /**
+   * Return a boolean to facilitate streaming elements on the basis of given predicate.
+   *
+   * @param entitiesFilter contain entities that should be attached to the FeatureSet
+   * @return boolean True if FeatureSet contains all entities in the entitiesFilter
+   */
+  public boolean hasAllEntities(List<String> entitiesFilter) {
+    List<String> allEntitiesName =
+        this.entities.stream()
+            .map(entitySpec -> entitySpec.toProto().getName())
+            .collect(Collectors.toList());
+    for (String entity : entitiesFilter) {
+      if (!allEntitiesName.contains(entity)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns a map of Feature references and Features if FeatureSet's Feature contains all labels in
+   * the labelsFilter
+   *
+   * @param labelsFilter contain labels that should be attached to FeatureSet's features
+   * @return Map of Feature references and Features
+   */
+  public Map<String, Feature> getFeaturesByAllLabels(Map<String, String> labelsFilter) {
```

Comment:
It seems strange that filtering is a part of this method. Can that be done externally with a static method? I would rather this be `getFeaturesByRef` and then have a filtering step outside.